tests/cmake/app_version/src/main.c

```diff
@@ -36,4 +38,39 @@ ZTEST(app_version, test_basic_strings)
 	zassert_equal(0, strcmp("5.6.7+89", APP_VERSION_TWEAK_STRING));
 }
 
+ZTEST(app_version, test_git_hash)
+{
+	/*
+	 * The git hashes in the two strings should be equal
+	 *
+	 * #define APP_GIT_COMMIT_HASH_SHORT    0x5c35660b
+	 * #define APP_BUILD_VERSION            v3.7.0-rc3-76-g5c35660b97bb
+	 *
+	 * Extract the numeric value from `APP_BUILD_VERSION`
+	 */
+	char build_version[] = STRINGIFY(APP_BUILD_VERSION);
+	int v_len = strlen(build_version);
+	unsigned int hash_from_string;
+	int g_idx = v_len;
+
+	/* Find the index of the `g` character, starting from the back */
+	while (g_idx-- > 0) {
+		if (build_version[g_idx] == 'g') {
+			break;
+		}
+	}
+	/* Should have terminated the loop early */
+	zassert_not_equal(0, g_idx);
+	/* Should be more than 8 characters between 'g' and end */
+	zassert_true((v_len - g_idx) > 8);
```

Comment:
Looks like you can make this test fail with `git config core.abbrev` but I guess that's OK!

Reply:
I don't believe so, I set `git config --global core.abbrev 4` and `git describe --abbrev=12 --always` still outputs a 12 character hash